### PR TITLE
support some show stmt

### PR DESCRIPTION
**Pull request by iamlinjunhong**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #4936 

## What this PR does / why we need it:

### CODE REVIEW

**Review by ouyuanning — APPROVED** [review]

**Review by yingfeng — APPROVED** [review]

**Review by daviszhen — APPROVED** [review]

File: pkg/common/moerr/error.go [hunk 1]
```diff
@@ -473,6 +473,10 @@ func GetOkExpectedEOF() *Error {
 	return &errOkExptededEOF
 }
 
+func New(code uint16, args ...any) *Error {
```
Comment (fengttt) [inline pkg/common/moerr/error.go:476]: Why do you need this one back?

Comment (fengttt) [reply pkg/common/moerr/error.go:476]: You should use Marshal/Unmarshal binary, if you need to transfer errors over wire.    Even though I don't think your code here need to transfer errors over wire -- maybe I missed something.

File: pkg/frontend/mysql_cmd_executor.go [hunk 2]
```diff
@@ -1116,6 +1116,44 @@ func (mce *MysqlCmdExecutor) handleSetVar(sv *tree.SetVar) error {
...
+	ses.Mrs.AddColumn(levelCol)
+	ses.Mrs.AddColumn(CodeCol)
+	ses.Mrs.AddColumn(MsgCol)
+
+	for i := len(ses.moerrs) - 1; i >= 0; i-- {
+		row := make([]interface{}, 3)
+		row[0] = "Error"
+		row[1] = ses.moerrs[i].ErrorCode()
+		row[2] = ses.moerrs[i].Error()
+		ses.Mrs.AddRow(row)
```
Comment (fengttt) [inline pkg/frontend/mysql_cmd_executor.go:1145]: Is this transferring error to clinet?     If it is last step, we may want to either include, or use the moerr.MySQLCode().

**Review by fengttt — COMMENTED** [review]
See inline comments about Marshal/Unmarshal errors and etc.   Thank you.

**Review by fengttt — COMMENTED** [review]
Do we really still maintain postgresql_sql.go?    I wonder what is the cost/benefit and if we should remove this thing all togehter.

File: pkg/frontend/mysql_protocol.go [hunk 3]
```diff
@@ -1494,6 +1494,12 @@ Error information includes several elements: an error code, SQLSTATE value, and
 	Message string: This string provides a textual description of the error.
 */
 func (mp *MysqlProtocolImpl) sendErrPacket(errorCode uint16, sqlState, errorMessage string) error {
+	if mp.ses != nil {
+		if len(mp.ses.moerrs) > MoDefaultErrorCount {
+			mp.ses.moerrs = mp.ses.moerrs[1:]
+		}
+		mp.ses.moerrs = append(mp.ses.moerrs, moerr.New(errorCode, errorMessage))
```
[the remark this answers is not in the document]
Comment (iamlinjunhong) [reply pkg/frontend/mysql_protocol.go]: The upper layer calls will process the moerr, and I need to get the error information finally passed to the client

File: pkg/frontend/mysql_cmd_executor.go (hunk 2 above)
Comment (iamlinjunhong) [reply pkg/frontend/mysql_cmd_executor.go:1147]: done

File: pkg/common/moerr/error.go (hunk 1 above)
Comment (iamlinjunhong) [reply pkg/common/moerr/error.go]: done

**Comment by iamlinjunhong** [comment]
> Do we really still maintain postgresql_sql.go? I wonder what is the cost/benefit and if we should remove this thing all togehter.

Currently, the mo is not maintained. However, some people in the community are interested in it. If it does not support multiple dialects in the future, deleting it will have no impact. And it will be deleted after community activities that add some mysql syntax.

**Review by fengttt — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Pull request merged**